packages/react/avatar/src/Avatar.test.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { axe } from 'jest-axe';
+import type { RenderResult } from '@testing-library/react';
+import { render, waitFor } from '@testing-library/react';
+import { getPartDataAttr } from '@radix-ui/utils';
+import * as Avatar from './Avatar';
+
+const ROOT_TEST_ID = 'avatar-root';
+const FALLBACK_TEXT = 'AB';
+const DELAY = 300;
+
+describe('given an Avatar with fallback and no image', () => {
+  let rendered: RenderResult;
+  let root: HTMLElement;
+  let fallback: HTMLElement;
+
+  beforeEach(() => {
+    rendered = render(
+      <Avatar.Root data-testid={ROOT_TEST_ID}>
+        <Avatar.Fallback>{FALLBACK_TEXT}</Avatar.Fallback>
+      </Avatar.Root>
+    );
+    root = rendered.getByTestId(ROOT_TEST_ID);
+    fallback = rendered.getByText(FALLBACK_TEXT);
+  });
+
+  it('should have no accessibility violations', async () => {
+    expect(await axe(rendered.container)).toHaveNoViolations();
+  });
+
+  it('should have a radix attribute on the root', () => {
+    const partDataAttr = getPartDataAttr('Avatar');
+    expect(root).toHaveAttribute(partDataAttr);
+  });
+
+  it('should have a radix attribute on the fallback', () => {
+    const partDataAttr = getPartDataAttr('AvatarFallback');
+    expect(fallback).toHaveAttribute(partDataAttr);
+  });
+});
+
+describe('given an Avatar with fallback and a working image', () => {
+  let rendered: RenderResult;
+
+  const orignalImage = window.Image;
+  beforeAll(() => {
+    jest.useFakeTimers();
+    (window.Image as any) = class MockImage {
+      onload: () => void = () => {};
+      src: string = '';
+      constructor() {
+        setTimeout(() => {
+          this.onload();
+        }, DELAY);
+        return this;
+      }
+    };
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+    window.Image = orignalImage;
+  });
+
+  beforeEach(() => {
+    rendered = render(
+      <Avatar.Root data-testid={ROOT_TEST_ID}>
+        <Avatar.Fallback>{FALLBACK_TEXT}</Avatar.Fallback>
+        <Avatar.Image data-testid="tst-123">{FALLBACK_TEXT}</Avatar.Image>
+      </Avatar.Root>
+    );
+  });
```

Comment:
🤔 img tags can't have children.

I think this should be :-

```jsx
  beforeEach(() => {
    rendered = render(
      <Avatar.Root data-testid={ROOT_TEST_ID}>
        <Avatar.Fallback>{FALLBACK_TEXT}</Avatar.Fallback>
        <Avatar.Image src="https://img.png" data-testid="tst-123" />
      </Avatar.Root>
    );
  });
```

Reply:
Oversight on my end 😅 thanks for catching!